examples/thumbs.rs

```diff
@@ -0,0 +1,69 @@
+//! Generate jpg thumbnails in parallel
+//!
+//! From <https://rust-lang-nursery.github.io/rust-cookbook/concurrency.html#generate-jpg-thumbnails-in-parallel>
+
+extern crate image;
+#[macro_use]
+extern crate quicli;
+extern crate structopt;
+
+use quicli::prelude::*;
+
+use std::path::{Path, PathBuf};
+use image::FilterType;
+
+/// Make some thumbnails
+#[derive(Debug, StructOpt)]
+struct Cli {
+    /// Which files?
+    #[structopt(default_value = "*.jpg")]
+    pattern: String,
+    /// How long should the longest edge of the thumbnail be?
+    #[structopt(long = "max-size", short = "s", default_value = "300")]
+    size: u32,
+    /// Where do you want to save the thumbnails?
+    #[structopt(long = "output", short = "o", default_value = "thumbnails")]
+    thumb_dir: String,
+    /// Pass many times for more log output
+    #[structopt(long = "verbosity", short = "v")]
+    verbosity: u64,
+}
+
+main!(|args: Cli, log_level: verbosity| {
+    let files: Vec<_> = glob(&args.pattern)?.filter_map(|x| x.ok()).collect();
+    create_dir(&args.thumb_dir)?;
+
+    info!("Saving {} thumbnails into {:?}...", files.len(), args.thumb_dir);
+
+    let image_failures: Vec<_> = files
+        .par_iter()
+        .map(|path| {
+            make_thumbnail(path, &args.thumb_dir, args.size).map_err(|e| {
```

Comment:
So, this is ugly. I want to:

- Print nice errors as soon as they happen
- count the errors later on (would require an AtomicI32 if done in a loop)